### PR TITLE
Store release publisher

### PR DESCRIPTION
**Pull request by dsdshcym**
First step of #753 

In this PR, I did following things:
1. Add `belongs_to publisher` relationship to `Release` schema
2. Update `Releases.publish/7` and related changeset functions to set/update `publisher` every time a release gets touched

A few questions:
1. Is it okay to just `cast :publisher_id` in the update changeset? I chose this impl over `put_assoc` because it's much more simpler and it doesn't require us to add `user` as arguments to `changeset/5`
2. Do we need to allow updating `publisher_id` after a release is created? If not, I'll move `cast :publisher_id` to create changeset.
3. Do I need to add migration script to backfill the `publisher_id` column now? Or I can do that after all the tasks in #753 are done?
4. Also, do we need to add `validate_required :publisher_id` to Release changesets? I chose not to do it because it caused several test cases in `ReleaseTest` to fail. And since we fully control how `params["publisher_id"]` gets passed in, the value of this validation seems to be little.

### CODE REVIEW

File: test/hexpm/repository/releases_test.exs [hunk 1]
```diff
@@ -42,6 +42,26 @@ defmodule Hexpm.Repository.ReleasesTest do
...
+      {:ok, %{release: release}} =
+        Releases.publish(
+          organization,
+          nil,
+          user,
+          "BODY",
+          meta,
+          "00",
+          audit: audit
+        )
```
Comment (ericmj) [inline test/hexpm/repository/releases_test.exs]: Can you avoid the long line break?

**Comment by ericmj** [comment]
1. I answered this in the review. We should only put external data in the params.
2. Yes, it should be updated to the last user to publish.
3. It would be a complicated migration so we will do that manually outside of migrations.
4. No need if we pass the user explicitly.

**Comment by wojtekmach** [comment]
Thank you!

**Comment by ericmj** [comment]
This was the script used to migrate the publisher_id from the audit logs:

```sql
UPDATE releases
  SET publisher_id = user_id
  FROM audit_logs
    WHERE action = 'release.publish'
      AND releases.id = (params->'release'->>'id')::integer;
```

**Pull request merged**